### PR TITLE
Update dev dependencies and transition scripts from CJS to ESM

### PR DESCRIPTION
**Pull request by ggetz**
Updates a few outdated npm dev dependencies as per [steps outlined in the Release guide](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/ReleaseGuide#one-week-before-release). 

`globby` in particular has [moved to pure ESM](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c). Since ESM can still import CommonJS packages, but CommonJS packages cannot import ESM packages synchronously, and that the general trend is moving towards ESM in general, this PR transitions `gulpfile.cjs`, `build.cjs`, and `server.cjs` to ESM. 

In doing so, I took the opportunity to clean up the following items:
* Uses updated way of defining tasks as [shown in the gulp documentation examples](https://gulpjs.com/docs/en/getting-started/creating-tasks)
* Uses `async/await` syntax throughout for flatter and more readable scrips
* Removes unneeded `bluebird` dependency
* Removes deprecated "request" dependency and replaces with the widely used and fairly minimal [`node-fetch`](https://www.npmjs.com/package/node-fetch) in `gulpfile.js`. (Potentially `server.js` could use [`http-proxy-middleware`](https://www.npmjs.com/package/http-proxy-middleware) but I wanted to limit the scope of this PR and determine if the local proxy is still needed).
* Uses updated way of defining tasks as [shown in the gulp documentation examples](https://gulpjs.com/docs/en/getting-started/creating-tasks)
* Used recommended new behavior with promises as recommended in the [karma docs](https://karma-runner.github.io/6.4/dev/public-api.html) in preparation for karma updates and to allow async/await syntax
* Added a `dry-run` parameter to upload script for testing purposes

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/gulp-esm/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: build.js [hunk 1]
```diff
@@ -577,81 +585,74 @@ const has_new_gallery_demos = ${newDemos.length > 0 ? "true;" : "false;"}\n`;
...
+  await writeFile(
+    path.join("Apps", "Sandcastle", "jsHintOptions.js"),
+    contents
+  );
+
+  return contents;
+}
+
+/**
+ * Bundles
```
Comment (sanjeetsuhag) [inline build.js:634]: Seems like the method descrition is incomplete.

File: server.js [hunk 2]
```diff
@@ -1,42 +1,39 @@
...
+import chokidar from "chokidar";
+import compression from "compression";
+import express from "express";
+import mkdirp from "mkdirp";
+import rimraf from "rimraf";
+import yargs from "yargs";
+
+const argv = yargs(process.argv)
+  .options({
+    port: {
```
Comment (sanjeetsuhag) [inline server.js:17]: On line 482, the error message for an incorrect port mentions `server.cjs` instead of `server.js`.

File: gulpfile.js [hunk 3]
```diff
@@ -169,61 +155,74 @@ function build(options) {
...
-  return build({
+  return buildCesium({
     minify: minify,
     removePragmas: removePragmas,
     sourcemap: sourcemap,
     node: node,
   });
-});
+}
+export default build;
```
Comment (sanjeetsuhag) [inline gulpfile.js:177]: Is this export needed?

File: gulpfile.js [hunk 4]
```diff
@@ -574,54 +554,86 @@ function isTravisPullRequest() {
...
+      },
+      directory: {
+        alias: "d",
+        description: "Upload directory.",
+        type: "string",
+        demandOption: true,
+      },
+      "cache-control": {
+        alias: "c",
+        description: "Only print file paths and S3 keys.",
```
Comment (sanjeetsuhag) [inline gulpfile.js:580]: The description needs to be updated.

**Review by sanjeetsuhag — COMMENTED** [review]
@ggetz This is mostly in good shape. Just needs a little more cleanup.

File: gulpfile.js (hunk 3 above)
Comment (ggetz) [reply gulpfile.js:177]: Yes, this becomes the [default gulp task](https://gulpjs.com/docs/en/getting-started/creating-tasks/).

**Comment by ggetz** [comment]
Thanks @sanjeetsuhag, updated!

**Comment by sanjeetsuhag** [comment]
Thanks @ggetz! `async`/`await` make the scripts much cleaner!

**Pull request merged**